src/main/java/org/opensearch/sdk/api/TransportActionsAPI.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk.api;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.extensions.ExtensionsOrchestrator;
+import org.opensearch.extensions.RegisterTransportActionsRequest;
+import org.opensearch.sdk.handlers.GenericResponseHandler;
+import org.opensearch.transport.TransportService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class helps manage transport actions for SDK
+ */
+public class TransportActionsAPI {
+    private final Logger logger = LogManager.getLogger(TransportActionsAPI.class);
+    private Map<String, Class> transportActions;
```

Comment:
(Suggestion) Can we be narrower on the type of `Class` here, to make sure developers are only registering the correct type?

Reply:
Sure, done.